### PR TITLE
Add default Optim to DPO example

### PR DESCRIPTION
**Pull request by natolambert**
Comparison here: https://wandb.ai/huggingface/huggingface/reports/RMSProp-DPO--Vmlldzo1Mzc2NDQ3

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Hi @natolambert, thanks for the PR. There is a lot of metric in the wandb report; which are the ones we should look at?


image

**Comment by natolambert** [comment]
Mostly this PR is just one of many things to get closer to the original implementation. Thanks for sharing the screenshot (I didn't realize it was private). 
@vwxyzjn the goal of this was to show _**there is not a big difference**_. We'll have more signal soon as we scale it.

File: examples/dpo.py [hunk 1]
```diff
@@ -145,6 +145,13 @@ def split_prompt_and_responses(sample) -> Dict[str, str]:
         report_to=script_args.report_to,
     )
 
+    # Trainer to match original paper
+    warmup_steps = 150
```
Comment (lewtun) [inline examples/dpo.py:149]: Should we make this configurable by the user?

Comment (tomaarsen) [reply examples/dpo.py:149]: Makes sense to me

Comment (natolambert) [reply examples/dpo.py]: fixed

**Review by younesbelkada — APPROVED** [review]
Thank you @natolambert !

**Pull request merged**